src/hotspot/share/oops/klass.hpp

```diff
@@ -213,6 +211,8 @@ class Klass : public Metadata {
 
   virtual bool is_klass() const { return true; }
 
+  bool is_in_klass_space() const { return !is_interface() && !is_abstract(); }
```

Comment:
I renamed this is_in_class_space() with the lower case 'c'.  It's still directing metaspace or indicating where the object was allocated.  Your name is a little better but I think not enough until we want to expand the things we want allocated in the class space.  As we talked about, with Tiny Class Pointers, class space will have different things in it (not that these new things need a compressed pointers).  But I think we're better off having less things in the space where their pointers can be compressed since this space is constrained.

Reply:
I have to think about this more.